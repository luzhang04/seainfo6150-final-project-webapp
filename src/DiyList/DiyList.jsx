import React from "react";
import PropTypes from "prop-types";
import styles from "./DiyList.module.css";
import DiyItem from "../DiyItem/DiyItem.jsx";

const DiyList = (props) => {
  let displayContent;

  if (props.crafts.length) {
    displayContent = (
      <ul className={styles.container}>
        {props.crafts.map((craft) => (
          <DiyItem craft={craft} key={craft.id} />
        ))}
      </ul>
    );
  } else {
    displayContent = <div>You have no data!</div>;
  }

  return (
    <div>
      {displayContent}
    </div>
  );
};

DiyList.propTypes = {
    crafts: PropTypes.array.isRequired
}
export default DiyList;
