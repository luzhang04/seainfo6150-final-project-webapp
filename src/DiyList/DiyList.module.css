.container {
  background: #ffe5d9;
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  gap: 20px;
  font-family: Arial, Helvetica, sans-serif;
  list-style-type: none;
  padding: 15px;
}
@media screen and (min-width: 200px) {
  .container {
    grid-template-columns: repeat(1, 1fr);
  }
}

@media screen and (min-width: 800px) {
  .container {
    grid-template-columns: repeat(2, 1fr);
  }
}

@media screen and (min-width: 1100px) {
  .container {
    grid-template-columns: repeat(3, 1fr);
  }
}

@media screen and (min-width: 1400px) {
  .container {
    grid-template-columns: repeat(4, 1fr);
  }
}
