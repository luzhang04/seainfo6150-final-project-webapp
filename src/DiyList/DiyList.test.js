import React from "react";
import { render } from "@testing-library/react";
import DiyList from "./DiyList.jsx";

describe("DiyList tests", () => {
    const crafts = [
      {"id": "0",
      "name" : "DIY Splatter Painted Easter Eggs",
      "category" : "easter",
      "description" : "Pick a color that'll match your décor, then create these chic splatter Easter eggs. Display them as a center piece or anywhere around the house.",
      "tutorial" : "<p>1. Make cake: Heat oven to 325°F. Butter 9- by 13-inch cake pan and line bottom with parchment, leaving overhang on 2 sides.</p><br/><p>2. In bowl, whisk together flour, baking powder, and salt.</p><br/><p>3. Using electric mixer, beat butter and granulated sugar on low speed to combine. Increase speed to medium-high; beat until light and fluffy, about 3 minutes. Reduce speed to medium; add eggs 1 at a time, beating until each is incorporated before adding next egg. Beat in vanilla.</p><br/><p>4. Reduce speed to low and add flour mixture in three parts, alternating with milk and beating just until incorporated.</p><br/><p>5. Spread batter in prepared pan and bake until wooden pick inserted in center comes out clean, about 40 minutes. Let cool in pan 10 minutes, then use overhangs to remove from pan and let cool completely.</p><br/><p>6. Make frosting: Using electric mixer, beat confectioners’ sugar, butter, and salt on low speed until combined, then increase speed to medium and beat until fluffy, about 3 minutes. With mixer running, slowly add cream. Beat until fluffy, about 3 minutes. Beat in vanilla.</p><br/><p>7. To decorate cake, cut out egg template and seven ¼-inch-wide strips of paper. Transfer cake to serving platter. Spread cake generously with buttercream. Place egg template in center of cake and score around outside; remove template.</p><br/><p>8. Use small offset spatula to smooth frosting inside scored egg shape. Place strips of paper on cake to create shapes inside egg. Working quickly with 1 section at a time, while icing is still wet, carefully fill each space with sprinkles, nonpareils, and sanding sugar. Place small pieces of plastic wrap over sections not being filled to keep colors from blending. Carefully peel away paper lines. (If frosting starts to crust over, brush lightly with water.)</p><br/>",
      "image": "https://hips.hearstapps.com/hmg-prod.s3.amazonaws.com/images/splatter-painted-easter-egg-designs-1616169765.jpg?crop=0.9514285714285714xw:1xh;center,top&resize=980:*",
      "timeStamp": "2018-11-22T17:55:57.000Z",
      "displayDate": "November 22nd 2018, 9:55 am"
      }
   ];

    it("renders correctly", () => {
      const { container } = render(<DiyList crafts={crafts}/>);
      expect(container).toMatchSnapshot();
    });
  });