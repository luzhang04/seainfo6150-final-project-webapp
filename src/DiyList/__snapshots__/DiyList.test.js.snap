// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiyList tests renders correctly 1`] = `
<div>
  <div>
    <ul
      class="container"
    >
      <li
        class="container"
      >
        <img
          alt="DIY Splatter Painted Easter Eggs"
          class="image"
          src="https://hips.hearstapps.com/hmg-prod.s3.amazonaws.com/images/splatter-painted-easter-egg-designs-1616169765.jpg?crop=0.9514285714285714xw:1xh;center,top&resize=980:*"
        />
        <header
          class="header"
        >
          <h2>
            <a
              class="link"
              href="/diylist/0"
            >
              DIY Splatter Painted Easter Eggs
            </a>
          </h2>
          <div
            class="content"
          >
            <time
              class="time"
              datetime="2018-11-22T17:55:57.000Z"
            >
              November 22nd 2018, 9:55 am
            </time>
            <p>
              Pick a color that'll match your d√©cor, then create these chic splatter Easter eggs. Display them as a center piece or anywhere around the house.
            </p>
          </div>
        </header>
      </li>
    </ul>
  </div>
</div>
`;
